tox.ini

```diff
@@ -76,10 +80,27 @@ deps =
     # test-requirements
     httpretty==0.9.6
     mock==1.3.0
-    pytest==2.8.7
     unittest2==1.1.0
     contextlib2==0.5.1
 
+[testenv:xenial]
+commands =
+  python ./tools/pipremove jsonschema
+  python -m pytest {posargs:tests/unittests cloudinit}
+basepython = python3
+deps =
+    # Refer to the comment in [xenial-shared-deps] for details
+    {[xenial-shared-deps]deps}
```

Comment:
i think tox-venv wont like this.
thats not really your problem, but unfortunate.

given you have *real* coverage of test in xenial (via package build),  maybe its file to just live with the '-dev' here ?

Reply:
Seems OK to me?

```
$ ./tools/tox-venv -h
Usage: tox-venv [--no-create] tox-environment [command [args]]
   run command with provided arguments in the provided tox environment
   command defaults to 'cmd' (see below).

   run with '--list' to show available environments

   if 'command' above is literal 'cmd' or '-', then the 'command' will
   be read from tox.ini.  This allows you to do:
      tox-venv py27 - tests/some/sub/dir
   and have the 'command' read correctly and have that execute:
      python -m pytest tests/some/sub/dir

environments in ./tox.ini
  py3*
  xenial-dev*
  pycodestyle*
  pyflakes*
  pylint*
  py27*
  doc*
  xenial*
  tip-pycodestyle
  tip-pyflakes
  tip-pylint
  citest*
$ ./tools/tox-venv xenial which python
inside tox:xenial running: which python
/home/daniel/dev/cloud-init/.tox/xenial/bin/python
$ ./tools/tox-venv xenial-dev which python
inside tox:xenial-dev running: which python
/home/daniel/dev/cloud-init/.tox/xenial-dev/bin/python
```

Reply:
good enough. thanks.